source/storing-data/provenance.rst

```diff
@@ -2,3 +2,352 @@ Decentralized Provenance Tracking
+      output-name: unweighted_unifrac_emperor
+      alias-of: 2adb9f00-a692-411d-8dd3-a6d07fc80a01
+
+- The type field describes the *type of the Action*: a :term:`Method`, :term:`Visualizer`, or :term:`Pipeline`.
+- The plugin field describes the plugin which registered the Action, details about which can be found in ``action.yaml``'s ``environment:plugins`` section. ``!ref`` is a custom YAML tag defined `here <https://github.com/qiime2/qiime2/blob/6d8932eda130d4a9356f977fece2e252c135d0b9/qiime2/core/archive/provenance.py#L84>`_, Generally, these custom tags provide a way to express a structure not easily described by basic YAML.
+- Inputs lists the registered names of all :term:`inputs<Input>` to the Action, as well as the UUIDs of the passed inputs. Note the distinction between inputs and parameters.
+- Parameters lists registered parameter names, and the user-passed (or selected default) values.
+- ``output-name`` is the user-passed name of the file written by this Action.
+- ``alias-of``: an optional field, present if the Action was run as part of a QIIME 2 :term:`Pipeline`, representing the _actual_ Action UUID rather than the Pipeline Alias. See maintainer note above for details.
+
+
+The environment block
+`````````````````````
+A non-comprehensive description of the computing environment in which this Action was run.
+It is not uncommon for QIIME 2 analyses to be run through multiple user interfaces, on multiple systems.
+For this reason, per-Action logging of system characteristics is useful.
+
+- ``platform``: the operating system and version used to run the Action. For VMs, this is the client OS.
+- ``python``: python version details, as captured by ``sys.version``
+- ``framework``: details about the QIIME 2 version used to performt this Action
+- ``plugin``: the QIIME 2 plugin, its version, and registered source web site
+- ``python-packages``: package names and version numbers for all packages in the global ``working_set`` of the active Python distribution, as collected by `pkg_resources <https://setuptools.readthedocs.io/en/latest/pkg_resources.html#workingset-objects>`_.
+
+.. admonition:: Maintainer Note
+   :class: maintainer-note
+
+   QIIME 2 currently captures only Python packages data, but we plan to expand this
+   to include all relevant packages in the environment regardless of language.
+   See the `github issue <http://github.com/qiime2/qiime2/issues/587>`_ if you are interested in contributing.
+
+.. code-block:: YAML
+
+   environment:
+      platform: macosx-10.9-x86_64
+      python: |-
+          3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:12:38)
+          [Clang 11.0.1 ]
+      framework:
+          version: 2021.4.0
+          website: https://qiime2.org
+          citations:
+          - !cite 'framework|qiime2:2021.4.0|0'
+      plugins:
+          diversity:
+              version: 2021.4.0
+              website: https://github.com/qiime2/q2-diversity
+      python-packages:
```

Comment:
Perhaps this list can be abbreviated a bit? I think a handful of lines gets the message across.